<a href="https://colab.research.google.com/github/anabiarochar/Mestrado_AlgoritmosProgramacao/blob/main/Projeto1/1_Bolha2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##Projeto 1 - Algoritmo de Ordenação Bolha (Bubble Sort)

Abaixo estão os algorítmos para ordenação, utilizando o método bolha de ordenação, com entradas de vetores de tamanhos 1.000, 10.000 e 100.000
- aleatórias
- ordenadas em forma crescente
- ordenadas em forma decrescente

Abaixo de cada algorítmo estão os tempos observados somando 4 tentativas.

##Vetor em ordem aleatória com 1000 elementos:



In [ ]:
import requests
import time

# Importar dados
url = "https://raw.githubusercontent.com/anabiarochar/Mestrado_AlgoritmosProgramacao/main/Projeto1/Vetores/vetor_1000_aleatorio.txt"

try:
    response = requests.get(url)
    content = response.text

    # Ler arquivo como vetor separado por vírgula
    alist = [int(x.strip()) for x in content.split(',') if x.strip()]

except Exception as e:
    print(f"Erro ao carregar ou processar o arquivo: {e}")

# Método Bolha
def bubbleSort(alist):
  for passnum in range(len(alist)-1,0,-1):
    for i in range(passnum):
      if alist[i]>alist[i+1]:
        temp = alist[i]
        alist[i] = alist[i+1]
        alist[i+1] = temp


# Execução e Tempo
if alist:
    inicio = time.perf_counter()
    vetor_ordenado = bubbleSort(alist)
    fim = time.perf_counter()

    tempo_total = fim - inicio

    print(f"Tempo de execução: {tempo_total:.6f} segundos")
    print(alist[:10])

else:
    print("Vetor vazio")

Tempo de execução: 0.050899 segundos
[1, 3, 8, 9, 9, 10, 10, 10, 10, 12]


###Tempo de Execução:

Tentativa 1 - Tempo de execução: 0.083225 segundos   
Tentativa 2 - Tempo de execução: 0.051019 segundos  
Tentativa 3 - Tempo de execução: 0.050137 segundos  
Tentativa 4 - Tempo de execução: 0.050899 segundos

##Vetor em ordem aleatória com 10000 elementos:

In [ ]:
import requests
import time

# Importar dados
url = "https://raw.githubusercontent.com/anabiarochar/Mestrado_AlgoritmosProgramacao/main/Projeto1/Vetores/vetor_10000_aleatorio.txt"

try:
    response = requests.get(url)
    content = response.text

    # Ler arquivo como vetor separado por vírgula
    alist = [int(x.strip()) for x in content.split(',') if x.strip()]

except Exception as e:
    print(f"Erro ao carregar ou processar o arquivo: {e}")

# Método Bolha
def bubbleSort(alist):
  for passnum in range(len(alist)-1,0,-1):
    for i in range(passnum):
      if alist[i]>alist[i+1]:
        temp = alist[i]
        alist[i] = alist[i+1]
        alist[i+1] = temp


# Execução e Tempo
if alist:
    inicio = time.perf_counter()
    vetor_ordenado = bubbleSort(alist)
    fim = time.perf_counter()

    tempo_total = fim - inicio

    print(f"Tempo de execução: {tempo_total:.6f} segundos")
    print(alist[:10])

else:
    print("Vetor vazio")

Tempo de execução: 7.299729 segundos
[3, 3, 4, 8, 9, 9, 9, 11, 12, 12]


###Tempo de Execução:

Tentativa 1 - Tempo de execução: 6.349090 segundos   
Tentativa 2 - Tempo de execução: 6.380606 segundos  
Tentativa 3 - Tempo de execução: 7.478226 segundos  
Tentativa 4 - Tempo de execução: 7.299729 segundos

##Vetor em ordem aleatória com 100000 elementos:

In [ ]:
import requests
import time

# Importar dados
url = "https://raw.githubusercontent.com/anabiarochar/Mestrado_AlgoritmosProgramacao/main/Projeto1/Vetores/vetor_100000_aleatorio.txt"

try:
    response = requests.get(url)
    content = response.text

    # Ler arquivo como vetor separado por vírgula
    alist = [int(x.strip()) for x in content.split(',') if x.strip()]

except Exception as e:
    print(f"Erro ao carregar ou processar o arquivo: {e}")

# Método Bolha
def bubbleSort(alist):
  for passnum in range(len(alist)-1,0,-1):
    for i in range(passnum):
      if alist[i]>alist[i+1]:
        temp = alist[i]
        alist[i] = alist[i+1]
        alist[i+1] = temp


# Execução e Tempo
if alist:
    inicio = time.perf_counter()
    vetor_ordenado = bubbleSort(alist)
    fim = time.perf_counter()

    tempo_total = fim - inicio

    print(f"Tempo de execução: {tempo_total:.6f} segundos")
    print(alist[:10])

else:
    print("Vetor vazio")

Tempo de execução: 776.809246 segundos
[2, 5, 6, 6, 8, 9, 10, 13, 14, 14]


###Tempo de Execução:

Tentativa 1 - Tempo de execução: 776.446981 segundos  
Tentativa 2 - Tempo de execução: 765.963112 segundos  
Tentativa 3 - Tempo de execução: 735.920505 segundos  
Tentativa 4 - Tempo de execução: 776.809246  segundos

##Vetor em ordem crescente com 1000 elementos:

In [15]:
import requests
import time

# Importar dados
url = "https://raw.githubusercontent.com/anabiarochar/Mestrado_AlgoritmosProgramacao/main/Projeto1/Vetores/vetor_1000_crescente.txt"

try:
    response = requests.get(url)
    content = response.text

    # Ler arquivo como vetor separado por vírgula
    alist = [int(x.strip()) for x in content.split(',') if x.strip()]

except Exception as e:
    print(f"Erro ao carregar ou processar o arquivo: {e}")

# Método Bolha
def bubbleSort(alist):
  for passnum in range(len(alist)-1,0,-1):
    for i in range(passnum):
      if alist[i]>alist[i+1]:
        temp = alist[i]
        alist[i] = alist[i+1]
        alist[i+1] = temp


# Execução e Tempo
if alist:
    inicio = time.perf_counter()
    vetor_ordenado = bubbleSort(alist)
    fim = time.perf_counter()

    tempo_total = fim - inicio

    print(f"Tempo de execução: {tempo_total:.6f} segundos")
    print(alist[:10])

else:
    print("Vetor vazio")

Tempo de execução: 0.017929 segundos
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]


###Tempo de Execução:

Tentativa 1 - Tempo de execução: 0.018424 segundos   
Tentativa 2 - Tempo de execução: 0.029141 segundos  
Tentativa 3 - Tempo de execução: 0.030002 segundos  
Tentativa 4 - Tempo de execução: 0.017929 segundos

##Vetor em ordem crescente com 10000 elementos:

In [19]:
import requests
import time

# Importar dados
url = "https://raw.githubusercontent.com/anabiarochar/Mestrado_AlgoritmosProgramacao/main/Projeto1/Vetores/vetor_10000_crescente.txt"

try:
    response = requests.get(url)
    content = response.text

    # Ler arquivo como vetor separado por vírgula
    alist = [int(x.strip()) for x in content.split(',') if x.strip()]

except Exception as e:
    print(f"Erro ao carregar ou processar o arquivo: {e}")

# Método Bolha
def bubbleSort(alist):
  for passnum in range(len(alist)-1,0,-1):
    for i in range(passnum):
      if alist[i]>alist[i+1]:
        temp = alist[i]
        alist[i] = alist[i+1]
        alist[i+1] = temp


# Execução e Tempo
if alist:
    inicio = time.perf_counter()
    vetor_ordenado = bubbleSort(alist)
    fim = time.perf_counter()

    tempo_total = fim - inicio

    print(f"Tempo de execução: {tempo_total:.6f} segundos")
    print(alist[:10])

else:
    print("Vetor vazio")

Tempo de execução: 2.323396 segundos
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]


###Tempo de Execução:

Tentativa 1 - Tempo de execução: 2.651374 segundos   
Tentativa 2 - Tempo de execução: 2.341335 segundos  
Tentativa 3 - Tempo de execução: 2.393616 segundos  
Tentativa 4 - Tempo de execução: 2.323396 segundos

##Vetor em ordem crescente com 100000 elementos:

In [21]:
import requests
import time

# Importar dados
url = "https://raw.githubusercontent.com/anabiarochar/Mestrado_AlgoritmosProgramacao/main/Projeto1/Vetores/vetor_100000_crescente.txt"

try:
    response = requests.get(url)
    content = response.text

    # Ler arquivo como vetor separado por vírgula
    alist = [int(x.strip()) for x in content.split(',') if x.strip()]

except Exception as e:
    print(f"Erro ao carregar ou processar o arquivo: {e}")

# Método Bolha
def bubbleSort(alist):
  for passnum in range(len(alist)-1,0,-1):
    for i in range(passnum):
      if alist[i]>alist[i+1]:
        temp = alist[i]
        alist[i] = alist[i+1]
        alist[i+1] = temp


# Execução e Tempo
if alist:
    inicio = time.perf_counter()
    vetor_ordenado = bubbleSort(alist)
    fim = time.perf_counter()

    tempo_total = fim - inicio

    print(f"Tempo de execução: {tempo_total:.6f} segundos")
    print(alist[:10])

else:
    print("Vetor vazio")

Tempo de execução: 231.035960 segundos
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]


###Tempo de Execução:

Tentativa 1 - Tempo de execução: 234.411430 segundos   
Tentativa 2 - Tempo de execução: 231.035960 segundos  
Tentativa 3 - Tempo de execução: 232.010231 segundos  
Tentativa 4 - Tempo de execução: 235.201392 segundos

##Vetor em ordem decrescente com 1000 elementos:

In [ ]:
import requests
import time

# Importar dados
url = "https://raw.githubusercontent.com/anabiarochar/Mestrado_AlgoritmosProgramacao/main/Projeto1/Vetores/vetor_1000_decrescente.txt"

try:
    response = requests.get(url)
    content = response.text

    # Ler arquivo como vetor separado por vírgula
    alist = [int(x.strip()) for x in content.split(',') if x.strip()]

except Exception as e:
    print(f"Erro ao carregar ou processar o arquivo: {e}")

# Método Bolha
def bubbleSort(alist):
  for passnum in range(len(alist)-1,0,-1):
    for i in range(passnum):
      if alist[i]>alist[i+1]:
        temp = alist[i]
        alist[i] = alist[i+1]
        alist[i+1] = temp


# Execução e Tempo
if alist:
    inicio = time.perf_counter()
    vetor_ordenado = bubbleSort(alist)
    fim = time.perf_counter()

    tempo_total = fim - inicio

    print(f"Tempo de execução: {tempo_total:.6f} segundos")
    print(alist[:10])

else:
    print("Vetor vazio")

Tempo de execução: 0.062281 segundos
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]


###Tempo de Execução:

Tentativa 1 - Tempo de execução: 0.062020 segundos   
Tentativa 2 - Tempo de execução: 0.063244 segundos  
Tentativa 3 - Tempo de execução: 0.062244 segundos  
Tentativa 4 - Tempo de execução: 0.062281 segundos

##Vetor em ordem decrescente com 10000 elementos:

In [ ]:
import requests
import time

# Importar dados
url = "https://raw.githubusercontent.com/anabiarochar/Mestrado_AlgoritmosProgramacao/main/Projeto1/Vetores/vetor_10000_decrescente.txt"

try:
    response = requests.get(url)
    content = response.text

    # Ler arquivo como vetor separado por vírgula
    alist = [int(x.strip()) for x in content.split(',') if x.strip()]

except Exception as e:
    print(f"Erro ao carregar ou processar o arquivo: {e}")

# Método Bolha
def bubbleSort(alist):
  for passnum in range(len(alist)-1,0,-1):
    for i in range(passnum):
      if alist[i]>alist[i+1]:
        temp = alist[i]
        alist[i] = alist[i+1]
        alist[i+1] = temp


# Execução e Tempo
if alist:
    inicio = time.perf_counter()
    vetor_ordenado = bubbleSort(alist)
    fim = time.perf_counter()

    tempo_total = fim - inicio

    print(f"Tempo de execução: {tempo_total:.6f} segundos")
    print(alist[:10])

else:
    print("Vetor vazio")

Tempo de execução: 8.880975 segundos
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]


###Tempo de Execução:

Tentativa 1 - Tempo de execução: 8.906069 segundos   
Tentativa 2 - Tempo de execução: 7.438931 segundos  
Tentativa 3 - Tempo de execução: 8.888610 segundos  
Tentativa 4 - Tempo de execução: 8.880975 segundos

##Vetor em ordem decrescente com 100000 elementos:

In [ ]:
import requests
import time

# Importar dados
url = "https://raw.githubusercontent.com/anabiarochar/Mestrado_AlgoritmosProgramacao/main/Projeto1/Vetores/vetor_100000_decrescente.txt"

try:
    response = requests.get(url)
    content = response.text

    # Ler arquivo como vetor separado por vírgula
    alist = [int(x.strip()) for x in content.split(',') if x.strip()]

except Exception as e:
    print(f"Erro ao carregar ou processar o arquivo: {e}")

# Método Bolha
def bubbleSort(alist):
  for passnum in range(len(alist)-1,0,-1):
    for i in range(passnum):
      if alist[i]>alist[i+1]:
        temp = alist[i]
        alist[i] = alist[i+1]
        alist[i+1] = temp


# Execução e Tempo
if alist:
    inicio = time.perf_counter()
    vetor_ordenado = bubbleSort(alist)
    fim = time.perf_counter()

    tempo_total = fim - inicio

    print(f"Tempo de execução: {tempo_total:.6f} segundos")
    print(alist[:10])

else:
    print("Vetor vazio")

Tempo de execução: 863.781448 segundos
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]


###Tempo de Execução:

Tentativa 1 - Tempo de execução: 858.927049 segundos  
Tentativa 2 - Tempo de execução: 858.957241 segundos  
Tentativa 3 - Tempo de execução: 855.807456 segundos  
Tentativa 4 - Tempo de execução: 863.781448 segundos  

In [ ]:
!echo "--- CPU Info ---"
!lscpu | grep "Model name"
!echo "--- RAM Info ---"
!free -h

--- CPU Info ---
Model name:                              Intel(R) Xeon(R) CPU @ 2.20GHz
--- RAM Info ---
               total        used        free      shared  buff/cache   available
Mem:            12Gi       818Mi       8.7Gi       1.0Mi       3.2Gi        11Gi
Swap:             0B          0B          0B
